src/jrd/inf.cpp

```diff
@@ -1166,6 +1166,10 @@ void INF_transaction_info(const jrd_tra* transaction,
 			memcpy(buffer, transaction->tra_attachment->att_database->dbb_database_name.c_str(), length);
 			break;
 
+		case fb_info_tra_snapshot_cn:
+			length = INF_convert(static_cast<SINT64>(transaction->tra_snapshot_number), buffer);
+			break;
+
```

Comment:
Seems we have misunderstanding here. At fb-devel we agree that:

>>   I also suggest that new isc_info_tra_snapshot_number returns the same data as SNAPSHOT_CN variable.

> Yes, no problem. But for RC it will be NULL as there is no request.

but code above returns 0 for every kind if RC transaction, while SNAPSHOT_CN variable returns request snapshot number, if available.

Reply:
Transaction info API have no request, so the request snapshot cannot be get. I did returned 0 meaning NULL in this context. How would you want to return NULL on transaction info?

Reply:
> Transaction info API have no request, so the request snapshot cannot be get

Yes. But when we use the same naming (fb_info_tra_**snapshot_cn**, **SNAPSHOT_CN**) it suggests that both ways is the equal. Thus i propose to change `fb_info_tra_snapshot_cn` by `fb_info_tra_snapshot_number`.

As for 0 instead of NULL - no objection of course.